apisix/init.lua

```diff
@@ -229,7 +230,11 @@ local function set_upstream_headers(api_ctx, picked_server)
 
     local hdr = core.request.header(api_ctx, "X-Forwarded-Proto")
     if hdr then
-        api_ctx.var.var_x_forwarded_proto = hdr
+        if type(hdr) == "table" then
+            api_ctx.var.var_x_forwarded_proto = core.table.concat(hdr, ", ")
```

Comment:
It doesn't use `,` as the separator, see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto

Reply:
So, X-Forwarded-Proto should be unique, not multiple?